src/stores/HistoryStore.js

```diff
@@ -0,0 +1,239 @@
+import { observable, action, reaction, runInAction } from 'mobx';
+import { filter } from 'lodash';
+import { TransactionStatus, Routes } from 'constants';
+import { transactions } from '../network/graphql/queries';
+
+const EVENT_HISTORY_LIMIT = 5;
+const ACTIVITY_HISTORY_LIMIT = 10;
+const INIT_VALUES = {
+  transactions: [],
+  myTransactions: [],
+  skip: 0, // skip for transaction
+  loaded: false,
+  loadingMore: false,
+  hasMore: true,
+  eventSkip: 0, // skip when query events
+  betSkip: 0,
+  resultSetSkip: 0,
+  withdrawSkip: 0,
+  limit: 0,
+};
+
+export default class {
+  @observable transactions = INIT_VALUES.transactions; // Full txs list fetched by batches
+  @observable myTransactions = INIT_VALUES.myTransactions; // Full txs list fetched by batches
+  @observable loadingMore = INIT_VALUES.loadingMore;
+  @observable loaded = INIT_VALUES.loaded;
+  @observable hasMore = INIT_VALUES.hasMore;
+  @observable myHasMore = INIT_VALUES.hasMore;
+  limit = INIT_VALUES.limit;
+  // for All Transaction in Event page and in Activity history page
+  skip = INIT_VALUES.skip;
+  eventSkip = INIT_VALUES.eventSkip;
+  betSkip = INIT_VALUES.betSkip;
+  resultSetSkip = INIT_VALUES.resultSetSkip;
+  withdrawSkip = INIT_VALUES.withdrawSkip;
+  // for My Transaction in Event page
+  mySkip = INIT_VALUES.skip;
+  myEventSkip = INIT_VALUES.eventSkip;
+  myBetSkip = INIT_VALUES.betSkip;
+  myResultSetSkip = INIT_VALUES.resultSetSkip;
+  myWithdrawSkip = INIT_VALUES.withdrawSkip;
+
+  constructor(app) {
+    this.app = app;
+
+    // Wallet addresses changed
+    reaction(
+      () => this.app.naka.account,
+      () => {
+        if (this.app.ui.location === Routes.ACTIVITY_HISTORY) {
+          this.init();
+        }
+      }
+    );
+    // New block
+    reaction(
+      () => this.app.global.syncBlockNum,
+      async () => {
+        if (this.transactions.length > 0 && this.app.ui.location === Routes.ACTIVITY_HISTORY) {
+          const { naka: { account } } = this.app;
+          const filters = { transactorAddress: account };
+          const newTxs = await this.fetchHistory(filters, ACTIVITY_HISTORY_LIMIT, 0, 0, 0, 0, 0, true);
+          const old = this.transactions[0];
+          if (newTxs.length > 0 && (newTxs[0].txid !== old.txid
```

Comment:
why would this happen? if we are skipping the correct numbers it should always return the next item? sounds like a backend bug.

Reply:
Oh this piece of codes is not related to the loadmore transaction.

When user creates an event, it goes to `my activity history` page to let the user see he just did this transaction.
This is a temp solution to update the first tx (since the user can create event one at time) from pending to success without refreshing the whole page.

Reply:
Do you have better idea? Since we also need similar solutions for event detail page, but that can be multiple txs.

Reply:
Ah I see. A more full-proof solution would prob be to do a find() on the transactions by txid and replace that index with the confirmed one. Can you always guarantee that it will be at index 0?

Reply:
Ya having multiple pending txs that need to be replaced should be done with a find() then cause I dont think the query will guarantee the index.

Reply:
I believe the pending is always at the top? So if thats the case, when you load more, you can check the pending items one by one, if you find that pending items in the confirmed items list, then you can remove it from the pending list.